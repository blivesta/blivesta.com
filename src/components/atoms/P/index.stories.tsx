import React from 'react';
import styled from 'styled-components';
import { storiesOf } from '@storybook/react';
import P from '.';

const Wrapper = styled.div`
  padding: 60px;
`;

storiesOf(`atoms|P`, module)
  .add(`default`, () => (
    <Wrapper>
      <P>
        この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。
      </P>
    </Wrapper>
  ))
  .add(`ja`, () => (
    <Wrapper>
      <P lang="ja">
        この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。
      </P>
    </Wrapper>
  ))
  .add(`en`, () => (
    <Wrapper>
      <P lang="en">
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore
        magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
        consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
        pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est
        laborum.
      </P>
    </Wrapper>
  ))
  .add(`justify`, () => (
    <Wrapper>
      <P lang="en" justify>
        Lorem ipsum dolor sit amet, consectetur adipisicing elit, sed do eiusmod tempor incididunt ut labore et dolore
        magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
        consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla
        pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est
        laborum.
      </P>
    </Wrapper>
  ));
