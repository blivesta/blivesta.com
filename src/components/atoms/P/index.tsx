import * as React from 'react';
import styled, { css } from 'styled-components';

const en = css`
  font-family: ${({ theme }) => theme.fonts.en};
  font-size: ${({ theme }) => theme.fontSizes[2]};
`;

const ja = css`
  font-family: ${({ theme }) => theme.fonts.ja};
  font-size: ${({ theme }) => theme.fontSizes[1]};
  line-height: 1.7;
`;

const StyledP = styled.p<PProps>`
  margin-top: 0;
  margin-bottom: ${({ theme }) => theme.spaces[3]};

  ${({ lang }) => (lang === 'en' ? en : ja)}
  ${({ justify }) =>
    justify &&
    css`
      text-align: justify;
    `}

  ${props =>
    props.last &&
    css`
      margin-bottom: ${props.theme.spaces[6]};
    `}
`;

export interface PProps {
  children: React.ReactNode;
  lang?: 'en' | 'ja';
  justify?: boolean;
  last?: boolean;
}

const P = ({ children, lang, justify, last }: PProps) => (
  <StyledP lang={lang} justify={justify} last={last}>
    {children}
  </StyledP>
);

export default P;
