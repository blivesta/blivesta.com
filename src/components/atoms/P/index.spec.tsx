import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import P from '.';

describe('P', () => {
  it('snapshot', () => {
    const tree = shallow(
      <P>この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。</P>,
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
