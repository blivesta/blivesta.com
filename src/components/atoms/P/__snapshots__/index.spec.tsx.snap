// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`P snapshot 1`] = `
<styled.p>
  この文章はダミーです。文字の大きさ、量、字間、行間等を確認するために入れています。この文章はダミーです。
</styled.p>
`;
